tools/contrib/modify_singlept_site_neon.py

```diff
@@ -0,0 +1,435 @@
+#! /usr/bin/env python
+"""
+|------------------------------------------------------------------|
+|---------------------  Instructions  -----------------------------|
+|------------------------------------------------------------------|
+This script is for modifying surface dataset at neon sites
+using data available from the neon server.
+
+After creating a single point surface data file from a global 
+surface data file using subset_data.py, use this script to 
+overwrite some fields with site-specific data for neon sites.
+
+This script will do the following:
+- Download neon data for the specified site if it does not exist 
+    in the specified directory.
+- Modify surface dataset with downloaded data (neon-specific).
+
+-------------------------------------------------------------------
+Instructions for running on Cheyenne/Casper:
+
+load the following into your local environment
+    module load python
+    ncar_pylib
+
+To remove NPL from your environment on Cheyenne/Casper:
+    deactivate
+-------------------------------------------------------------------
+To see the available options:
+    ./modify_singlept_site_neon.py --help
+-------------------------------------------------------------------
+"""
+# TODO
+# if file not found run subset_data.py
+# Clean up imports for both codes...
+# Check against a list of valid names.
+
+#  Import libraries
+from __future__ import print_function
+
+import sys
+import os
+import glob
+import argparse
+import requests
+
+import numpy as np
+import pandas as pd
+import xarray as xr
```

Comment:
@wwieder these are all standard python packages that you get with ncar_pylib on cheyenne. There are two aspects to python environments, one is the list of python modules used in the python scripts, the other is in setting up the python environment that includes setting specific versions of those. The first is in the script itself, and the only thing that you can do is to check for a specific version of python and/or a python module. It would be good to at least add a check for the python version (this is something we do in cime for example). If you use something version specific in a python module you should do the same. I'm guessing that for this list of modules we aren't using anything version specific.

In terms of setting up the python environment that's outside the script. I looked and didn't see anything specific that CESMLab is doing in this regard, nor are they doing version checking inside the scripts (at least not yet). But, as cheyenne is our workhorse and ncar_pylib is sufficient I think this is fine as it is right now.

I will add an issue to have python scripts check for versions though.

Reply:
Thanks @ekluzek, I mostly want to make sure the script works, even as ncar_pylib gets updated.